mindsdb/integrations/handlers/statsforecast_handler/statsforecast_handler.py

```diff
@@ -0,0 +1,102 @@
+import pandas as pd
+import dill
+from mindsdb.integrations.libs.base import BaseMLEngine
+from statsforecast import StatsForecast
+from statsforecast.models import AutoARIMA
+
+DEFAULT_FREQUENCY = "D"
+DEFAULT_MODEL = AutoARIMA()
+DEFAULT_MODEL_NAME = "AutoARIMA"
+
+
+class StatsForecastHandler(BaseMLEngine):
+    """Integration with the Nixtla StatsForecast library for
+    time series forecasting with classical methods.
+    """
+
+    name = "statsforecast"
+
+    def create(self, target, df=None, args={}):
+        """Create the StatsForecast Handler.
+
+        Requires specifying the target column to predict and time series arguments for
+        prediction horizon, time column (order by) and grouping column(s).
+
+        Does not require a dataframe, because model fitting takes place jointly with
+        forecasting in the "predict step".
+        """
+        time_settings = args["timeseries_settings"]
+        assert time_settings["is_timeseries"], "Specify time series settings in your query"
+        ###### store model args and time series settings in the model folder
+        model_args = {}
+        model_args["target"] = target
+        model_args["horizon"] = time_settings["horizon"]
+        model_args["order_by"] = time_settings["order_by"]
+        model_args["group_by"] = time_settings["group_by"]
+
+        ###### persist changes to handler folder
+        self.model_storage.file_set("model_args", dill.dumps(model_args))
+
+    def predict(self, df, args={}):
+        """Makes forecasts with the StatsForecast Handler.
+
+        This method takes care of both model fitting and forecasting in one step.
+        Model fitting is extremely fast, so model parameters do not need to be
+        stored separately. Instead, we re-fit each time this method is called.
```

Comment:
Any chance that a huge dataset could take longer? 

The training process is detached from the main one, so e.g. the GUI doesn't "freeze" until getting back an OK response, but the prediction process is synchronous, so I wonder if this is too aggressive.

Reply:
Very good spot!

I profiled a big dataset (300k rows) and you were right - model fitting took about a minute on my machine, which has 12 cores. Prediction was almost instant.

I'll make the necessary changes. It's a little bit hacky as prefers you to use the 'forecast()' method than 'fit()' then 'predict()', but in this case the hack seems defo worth it.

Nice catch again.